### PR TITLE
fix(ui): Long overdue - Fix red error screens during OIDC login, logout exception scenarios

### PR DESCRIPTION
**Pull request by jjoyce0510**
**Summary**

Previously, when we failed to login or logout using SSO (oidc), we hit a red screen of death. In this PR we've added logic to 

1. Clearly log the error to the admin / operator
2. Redirect the poor user back to the /login page with a persistent error warning pinned to the top. 

**Screenshots**
Below is the new error scenario, redirect to login with an error!

Screen Shot 2022-08-22 at 8 12 17 PM
Screen Shot 2022-08-22 at 7 47 35 PM

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-frontend/app/controllers/AuthenticationController.java [hunk 1]
```diff
@@ -275,8 +275,17 @@ private Result redirectToIdentityProvider() {
...
+
+        try {
+            final HttpAction action = client.redirect(playWebContext);
+            return new PlayHttpActionAdapter().adapt(action.getCode(), playWebContext);
+        } catch (Exception e) {
+            _logger.error("Caught exception while attempting to redirect to SSO identity provider! It's likely that SSO integration is mis-configured", e);
+            return redirect(
+                String.format("/login?error_msg=%s",
+                URLEncoder.encode("Failed to redirect to Single Sign-On provider. Please contact your DataHub Administrator, "
+                    + "or refer to server logs for more information.")));
```
Comment (chriscollins3456) [inline datahub-frontend/app/controllers/AuthenticationController.java:287]: just curious, why did you need to split this sentence up and add the two strings together?

**Review by chriscollins3456 — APPROVED** [review]
neato

File: datahub-frontend/app/controllers/AuthenticationController.java (hunk 1 above)
Comment (jjoyce0510) [reply datahub-frontend/app/controllers/AuthenticationController.java:287]: line got too long for lint!
